src/schedule/ucc_schedule.h

```diff
@@ -49,9 +51,10 @@ typedef struct ucc_coll_task {
     ucc_ev_t                    *ev;
     void                        *ee_task;
     ucc_coll_task_t             *triggered_task;
-    /* used for progress queue */
+    /* used for st & locked mt progress queue */
     ucc_list_link_t              list_elem;
-    uint8_t                      was_progressed;
+    /* used for lf mt progress queue */
+    ucc_lf_queue_elem_t          lf_elem;
```

Comment:
should we put list_elem and lf_elem under "union" ? they can't be used simultaneously.